### PR TITLE
Add Scine QCMaquis recipe to builtin packages

### PR DESCRIPTION
**Pull request by adam-grofe**
[Scine QCMaquis](https://scine.ethz.ch/download/qcmaquis) is a DMRG program developed by the Markus Reiher group at ETH. In this PR, I add a spack recipe for compiling the code.

### CODE REVIEW

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    depends_on("googletest+gmock", when="build_tests=True")
+
+    def cmake_args(self):
+        args = []
+        build_symm = str(self.spec.variants["symmetries"].value)
+        if build_symm:
+            args.extend(["-DBUILD_SYMMETRIES={}".format(build_symm)])
+        build_tests = str(self.spec.variants["build_tests"].value)
+        if build_tests:
+            args.extend(["-DQCMAQUIS_TESTS=On"])
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: Have you checked out the helper functions at https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake?

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        "blas",
+        values=("openblas", "mkl"),
+        default="openblas",
+        description="Which blas library to use.",
+    )
+    variant(
+        "symmetries",
+        default="TwoU1;SU2U1;SU2U1PG;TwoU1PG",
+        description='Wave functions symmetries to compile (e.g. "TwoU1;SU2U1;SU2U1PG;TwoU1PG")',
+    )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: Would it make sense for this package to make this a multi-valued variant that can take multiple values?  (See the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#multi-valued-variants.)

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 1 above)
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: This isn't doing what you want.  

Try `spack install scene-qcmaquis~build_tests` (so disable the variant) and you'll see that the option is translated to `'-DQCMAQUIS_TESTS=On'`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the homepage and three version branches; however, the variants appear to need work.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 1 above)
Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: Thank you for the suggestion; I made the suggested changes.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 2 above)
Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: I changed this variant to use the multi.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 1 above)
Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: After changing the cmake args function to use `define_from_variant`, this bug was fixed. I verified it myself in a clean environment.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        "blas",
+        values=("openblas", "mkl"),
+        default="openblas",
+        description="Which blas library to use.",
+    )
+    variant(
+        "symmetries",
+        default="SU2U1PG,TwoU1PG",
+        description='Wave functions symmetries to compile (e.g. "SU2U1PG,TwoU1PG")',
+        values=("U1", "TwoU1", "TwoU1PG", "NU1", "Z2", "SU2U1", "SU2U1PG", "U1DG", "NONE"),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py:30]: Should the value(s) be validated?  For example, what if someone included `NONE`?

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    root_cmakelists_dir = "dmrg"
+
+    depends_on("hdf5~mpi")
+    depends_on("lapack")
+    depends_on("openblas+ilp64 threads=openmp", when="blas=openblas")
+    depends_on("intel-oneapi-mkl", when="blas=mkl")
+    depends_on("gsl")
+    depends_on("boost+program_options+filesystem+system+thread+serialization+chrono")
+    depends_on("googletest+gmock", when="build_tests=True")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: While this works, conditional dependencies involving boolean variants usually use the `+` or `~` syntax (i.e., `when="+build_tests"`) as shown in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies.

**Review by tldahlgren — COMMENTED** [review]
Looking good.  Just have a couple of questions/pointers.

**Comment by tldahlgren** [comment]
Please ping me when you're ready for this to be merged.

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 3 above)
Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py:30]: According the QcMaquis CMakeLists.txt, "TwoU1PG and SU2U1PG are built in any case."

File: var/spack/repos/builtin/packages/scine-qcmaquis/package.py (hunk 4 above)
Comment (adam-grofe) [reply var/spack/repos/builtin/packages/scine-qcmaquis/package.py]: Thank you for the suggestion. I have made this change to the package.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**